frame/support/src/lib.rs

```diff
@@ -1645,6 +1645,9 @@ pub mod pallet_prelude {
 /// used using `#[pallet::compact]`, function must return `DispatchResultWithPostInfo` or
 /// `DispatchResult`.
 ///
+/// Each dispatchable may also be annotated with the `#[pallet::call_index($idx)]` attribute,
+/// which defines and sets the codec index for the dispatchable function in the `Call` enum.
```

Comment:
can you briefly add a description here of what happens if you use the call index in some placed, but not in others?

Can we make it so that call index must be used everywhere or nowhere?

Reply:
Honestly, probably call index should just be mandatory. We could allow users to ignore it if they use [`pallet::dev_mode`](https://github.com/paritytech/substrate/issues/8965)

Reply:
In the spirit of keeping the FRAME boilerplate minimal, I don't think mandating call-indices should be mandatory. I like how it works in `construct_runtime` and find it intuitive: You have the default order, and you can overwrite anything that you want. Duplicates will be detected at compile time.